common/unsafe/src/main/java/org/apache/spark/unsafe/array/ByteArrayMethods.java

```diff
@@ -47,13 +47,20 @@ public static int roundNumberOfBytesToNearestWord(int numBytes) {
   public static boolean arrayEquals(
       Object leftBase, long leftOffset, Object rightBase, long rightOffset, final long length) {
     int i = 0;
-    while (i <= length - 8) {
-      if (Platform.getLong(leftBase, leftOffset + i) !=
-        Platform.getLong(rightBase, rightOffset + i)) {
-        return false;
-      }
-      i += 8;
-    }
+
+      // This attempts to speed up the memcmp type of operation, but there is no way
+      // to guarantee that the offsets will be on a word boundary in order to use
+      // Platform.getLong
```

Comment:
It can still be used only if both the leftOffset AND rightOffset start on proper word boundaries.
By checking for these 2 conditions, we lose the advantage gained by this block.